ui/app/models/sync/destinations/vercel-project.js

```diff
@@ -6,11 +6,42 @@
 import SyncDestinationModel from '../destination';
 import { attr } from '@ember-data/model';
 import { withFormFields } from 'vault/decorators/model-form-fields';
+import { withModelValidations } from 'vault/decorators/model-validations';
 
-@withFormFields()
+const validations = {
+  name: [{ type: 'presence', message: 'Name is required.' }],
+  accessToken: [{ type: 'presence', message: 'Access token is required.' }],
+  projectId: [{ type: 'presence', message: 'Project ID is required.' }],
+  deploymentEnvironments: [{ type: 'presence', message: 'At least one environment is required.' }],
+};
+const fields = ['name', 'accessToken', 'projectId', 'teamId', 'deploymentEnvironments'];
+
+@withModelValidations(validations)
+@withFormFields(fields)
 export default class SyncDestinationsVercelProjectModel extends SyncDestinationModel {
-  @attr('string') accessToken;
-  @attr('string') projectId;
-  @attr('string') teamId;
-  @attr('array') deploymentEnvironments;
+  @attr('string', {
+    subText: 'Vercel API access token with the permissions to manage environment variables.',
+  })
+  accessToken;
+
+  @attr('string', {
+    label: 'Project ID',
+    subText: 'Project ID where to manage environment variables.',
+  })
+  projectId;
+
+  @attr('string', {
+    label: 'Team ID',
+    subText: 'Team ID the project belongs to. Optional.',
+  })
+  teamId;
+
+  // TODO can also be a string, return and test how this works with live API
```

Comment:
Go treats arrays and a list of strings as the same. The [docs](https://developer.hashicorp.com/vault/api-docs/system/secrets-sync#deployment_environments) call it a string type, but the sample payload shows an array. 

I opted to use an array because they're easier to manipulate in forms. Leaving the note here so we test with the real API and account for both data types in component tests